pkg/buildcontext/git.go

```diff
@@ -55,6 +61,18 @@ func (g *Git) UnpackTarFromBuildContext() (string, error) {
 	return directory, err
 }
 
+func getGitAuth() transport.AuthMethod {
+	username := os.Getenv(gitAuthUsernameEnvKey)
+	password := os.Getenv(gitAuthPasswordEnvKey)
+	if username != "" || password != "" {
```

Comment:
Thanks for adding this new feature. Would this work, if user only specified username but no password?
Shd it be 
1. both are present or 
2. None?

Reply:
Thanks for the review!

I set it as `||` because github handles the case where the username is the personal access token and the password is blank. In certain use cases, this is desirable to avoid the PAT being cached in the credentials helpers. FWIW, I do exactly that in my test example above in the PR description.

Reply:
Thanks for the explanation.